controllers/storageclient_controller.go

```diff
@@ -526,98 +456,63 @@ func (s *StorageClientReconciler) reconcileClientStatusReporterJob(instance *v1a
 		return nil
 	}
-
-	// try to list the default client who is the default storage claims owner
-	claimOwners := &v1alpha1.StorageClientList{}
-	if err := s.list(claimOwners, client.MatchingFields{defaultClaimsOwnerIndexName: "true"}); err != nil {
-		return fmt.Errorf("failed to list default storage claims owner: %v", err)
+	blockClaim := &v1alpha1.StorageClaim{}
+	blockClaim.Name = fmt.Sprintf("%s-ceph-rbd", r.storageClient.Name)
+	blockClaim.Spec.Type = "block"
+	blockClaim.Spec.StorageClient = &v1alpha1.StorageClientNamespacedName{
+		Name: r.storageClient.Name,
 	}
-
-	if len(claimOwners.Items) == 0 {
-		// no other storageclient claims as an owner and take responsibility of creating the default claims by becoming owner
-		if utils.AddAnnotation(instance, defaultClaimsOwnerAnnotationKey, "true") {
-			if err := s.update(instance); err != nil {
-				return fmt.Errorf("not able to claim ownership of creating default storageclaims: %v", err)
-			}
-		}
+	if err := r.own(blockClaim); err != nil {
+		return fmt.Errorf("failed to own storageclaim of type block: %v", err)
 	}
-
-	// we successfully took the ownership to create a default claim from this storageclient, so create default claims if not created
-	// after claiming as an owner no other storageclient will try to take ownership for creating default storageclaims
-	annotations := instance.GetAnnotations()
-	if annotations[defaultClaimsOwnerAnnotationKey] == "true" && annotations[defaultClaimsProcessedAnnotationKey] != "true" {
-		if err := s.createDefaultBlockStorageClaim(instance); err != nil && !kerrors.IsAlreadyExists(err) {
-			return fmt.Errorf("failed to create %q storageclaim: %v", defaultBlockStorageClaim, err)
-		}
-		if err := s.createDefaultSharedfileStorageClaim(instance); err != nil && !kerrors.IsAlreadyExists(err) {
-			return fmt.Errorf("failed to create %q storageclaim: %v", defaultSharedfileStorageClaim, err)
-		}
+	if err := r.create(blockClaim); err != nil && !kerrors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create block storageclaim: %v", err)
 	}
-
-	// annotate that we created default storageclaims successfully and will not retry
-	if utils.AddAnnotation(instance, defaultClaimsProcessedAnnotationKey, "true") {
-		if err := s.update(instance); err != nil {
-			return fmt.Errorf("not able to update annotation for creation of default storageclaims: %v", err)
-		}
-	}
-
 	return nil
 }
 
-func (s *StorageClientReconciler) createDefaultBlockStorageClaim(instance *v1alpha1.StorageClient) error {
-	storageclaim := &v1alpha1.StorageClaim{}
-	storageclaim.Name = defaultBlockStorageClaim
-	storageclaim.Spec.Type = "block"
-	storageclaim.Spec.StorageClient = &v1alpha1.StorageClientNamespacedName{
-		Name:      instance.Name,
-		Namespace: instance.Namespace,
+func (r *StorageClientReconciler) reconcileSharedfileStorageClaim() error {
+	if r.storageClient.GetAnnotations()[storageClaimProcessedAnnotationKey] == "true" {
```

Comment:
The annotation concern is added at a upper level. I would expect the guard to be outside of the reconciler function and not inside it.

Reply:
makes sense, will do.

Reply:
done.